src/nvim/ex_cmds2.c

```diff
@@ -2323,9 +2324,12 @@ void ex_scriptnames(exarg_T *eap)
 
   for (int i = 1; i <= script_items.ga_len && !got_int; i++) {
     if (SCRIPT_ITEM(i).sn_name != NULL) {
-      home_replace(NULL, SCRIPT_ITEM(i).sn_name,
-                   NameBuff, MAXPATHL, true);
-      smsg("%3d: %s", i, NameBuff);
+      home_replace(NULL, SCRIPT_ITEM(i).sn_name, NameBuff, MAXPATHL, true);
+      vim_snprintf((char *)IObuff, IOSIZE, "%3d: %s", i, NameBuff);
+      msg_putchar('\n');
+      msg_outtrans(IObuff);
+      ui_flush();  // output one line at a time
```

Comment:
does it work properly without `ui_flush()` ? ui_flush should already be emitted by message.c precisely when needed, while avoid unnecessary flicker.

Reply:
Removing `ui_flush()` indeed seems to still work properly and is faster.

Reply:
However, it seems `ui_flush()` is used in many places that output to messages.

Reply:
yes, and in many of those places it can likely be removed :)